chart/kubeapps/templates/db-secret-jobs-cleanup.yaml

```diff
@@ -1,3 +1,4 @@
+{{- if .Values.postgresql.existingSecret -}}
```

Comment:
I'm confused about this condition, but I think it's just the name. The problem is that the `.Values.postgresql.existingSecret` gets a default value, which means it's not really telling us if the user is using an existing secret?

In reality, I think the only time we *don't* want to delete this is if the user is using an existing secret that's been created out-of-band. But we don't have a way to know that currently.

Could we not improve this situation here by:

1. remove the current default for `existingSecret` so that it is only defined if explicitly done so by the user,
2. Don't use `kubeapps.postgresql.fullname` when `existingSecret` is not defined, but instead use the current default value of `kubeapps-db` (but perhaps define it as a helper constant so it's just defined once and can be updated with a major release if we want)

? I think with that, existing installs where `existingSecret` hasn't been explicitly defined will still use `kubeapps-db` when they upgrade. Those that have defined `existingSecret` will still continue to use that value. And importantly, we can then update the condition here to only delete the secret if `existingSecret` is undefined (ie. user hasn't explicitly set an existing secret, so we can delete the one kubeapps created).

Reply:
Yes, the name is confusing. The thing is that, by default, postgresql cannot be updated/upgraded. This is because, if left by default, the chart will create a secret with a random password. When upgraded, that secret will be generated again, overriding the password. You need to either, set a password manually or specify a secret. 

In kubeapps we are doing the second. We are generating a secret in the pre-install hook (so it's not modified in the future) and we are filling it with a random password (https://github.com/kubeapps/kubeapps/blob/master/chart/kubeapps/templates/db-secret-bootstrap.yaml). Because of that, we need to "manually" clean up that secret as well when uninstalling. 

We don't allow users to specify their own secret. The two options are: manually setting the password or letting it to be auto-generated.

Reply:
> Yes, the name is confusing. The thing is that, by default, postgresql cannot be updated/upgraded. This is because, if left by default, the chart will create a secret with a random password. When upgraded, that secret will be generated again, overriding the password. You need to either, set a password manually or specify a secret.
> 
> In kubeapps we are doing the second. We are generating a secret in the pre-install hook (so it's not modified in the future) and we are filling it with a random password (https://github.com/kubeapps/kubeapps/blob/master/chart/kubeapps/templates/db-secret-bootstrap.yaml). Because of that, we need to "manually" clean up that secret as well when uninstalling.

Yes, I think all of the above is what I'd understood - this is why the Kubeapps chart currently defaults `postgresql.existingSecret=kubeapps-db` which is confusing because `existingSecret` implies that the secret existed already, wherease Kubeapps is creating it during install.

What wasn't clear to me is that I think you're using `kubeapps.postgresql.fullname` for the secret when `existingSecret` is not defined because that'll be the secret which is created by the postgresql chart when setting `postgresql.postgresqlPassword` I think?

> 
> We don't allow users to specify their own secret. The two options are: manually setting the password or letting it to be auto-generated.

Right, and this is the specific cased raised in the issue - understood. I think it'd be much clearer if Kubeapps didn't use the `existingSecret` option to mean "even though there isn't an existing secret, we'll create one and tell postgres that it existed". An alternative would have been:

1. if (a new) `.Values.userExistingSecret` is unset, do the current behaviour of defaulting `postgresql.existingSecret=kubeapps-db` and clean it up afterwards
2. if `.Values.userExistingSecret` is set, use it for `postgresql.exstingSecret` and *don't* clean it up afterwards

With something like that, I think the issue would be solved with an existing secret, rather than specifying an actual password (though that would still be possible). Anyway, don't want to block on this.

Reply:
> 1. if (a new) `.Values.userExistingSecret` is unset, do the current behaviour of defaulting `postgresql.existingSecret=kubeapps-db` and clean it up afterwards
> 2. if `.Values.userExistingSecret` is set, use it for `postgresql.exstingSecret` and _don't_ clean it up afterwards

As far as I know, you cannot do this with Helm. You cannot set a value from other. You would need to use helpers for that but we cannot modify the existing helpers of the PostgreSQL chart (which is just reading `.Values.existingSecret`.